kore/src/Kore/Logger/Output.hs

```diff
@@ -207,7 +217,29 @@ filterSeverity level entry =
 
 -- | Run a 'LoggerT' with the given options.
 runLoggerT :: KoreLogOptions -> LoggerT IO a -> IO a
-runLoggerT options = withLogger options . runReaderT . getLoggerT
+runLoggerT options loggerT = do
+    let runLogger = runReaderT . getLoggerT $ loggerT
+    withLogger options $ \logger -> do
+        modifiedLogger <- concurrentLogger logger
+        runLogger modifiedLogger
+
+concurrentLogger :: LogAction IO a -> IO (LogAction IO a)
+concurrentLogger logger = do
+    tChan <- newTChanIO
+    _ <- forkIO $ do
+            val <- atomically $ readTChan tChan
+            logger Colog.<& val
+            return ()
```

Comment:
... following the suggestion above, we don't need to explicitly `return ()` here.
```suggestion
```